File: greedy/rust_greedy/src/LeetCode455.rs
struct Solution{
    
}

impl Solution {
    pub fn find_content_children(g: Vec<i32>, s: Vec<i32>) -> i32 {
        g.sort();
        s.sort();
        let mut child = 0;
        let mut cookie = 0;
        
        while child < g.len() && cookie < s.len() {
            if g[child] <= s[cookie]{
                child++;
            }
            cookie++;
        }
        return child;
    }
}

File: greedy/LeetCode455.cc
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std;

class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        if(g.size() == 0 || s.size() == 0){
            return 0;
        }
        sort(g.begin(),g.end());
        sort(s.begin(),s.end());
        int child = 0, cookie = 0;
        while(child < g.size() && cookie < s.size()){
            if(g[child] <= s[cookie]){
                child++;
            }
            cookie++;
        }
        return child;
    }
};